freezer/freezer_url.go

```diff
@@ -44,29 +45,36 @@ func newFreezerSink(u *url.URL) (substrate.AsyncMessageSink, error) {
 	}
 
 	var streamstore straw.StreamStore
+	var err error
 	switch u.Scheme {
 	case "freezer+dir":
-		streamstore = &straw.OsStreamStore{}
+		streamstore, err = strawOpen("file:///")
+		if err != nil {
+			return nil, err
+		}
 	case "freezer+s3":
-		var (
-			enc straw.S3Option
-			err error
-		)
-
-		sse := q.Get("sse")
-		switch sse {
-		case "":
-		case "aes256":
-			enc = straw.S3ServerSideEncoding(straw.ServerSideEncryptionTypeAES256)
-		default:
-			return nil, fmt.Errorf("unsupported value: %s passed for sse parameter", sse)
+		u1 := url.URL{
+			Scheme: "s3",
+			Host:   u.Host,
 		}
 
-		if enc != nil {
-			streamstore, err = straw.NewS3StreamStore(u.Hostname(), enc)
-		} else {
-			streamstore, err = straw.NewS3StreamStore(u.Hostname())
+		// carry through a whitelist of query params for the straw URL
+		newVals := url.Values(make(map[string][]string))
+		for k, vals := range u.Query() {
+			for _, val := range vals {
+				switch k {
+				case "sse":
+					// substrate has always allowed & expected lower case values such as
+					// `aes256` but straw expects upper case.  Convert here so that either
+					// works with substrate now.
+					newVals.Add(k, strings.ToUpper(val))
```

Comment:
I think it would be better to make both case insensitive rather than doing conversions like this

Reply:
Straw doesn't really expect one or the other, it just passes whatever it gets to to AWS, which (I believe) expects upper.  I'd rather not add more munging in straw which currently doesn't have any.  If I was doing substrate again, I'd probably make it upper-only from the start.  This code here is really just for compatibility.

Reply:
Fair, though I think we should still make substrate case insensitive, as then we will be able to use uppercase as in straw without breaking current users...

Reply:
I don't understand what you mean.  This change does make it case insensitive for substrate users.

Reply:
Ah yeah, sorry got myself confused 😅